src/main/java/com/microsoft/aad/msal4j/MSALRuntimeAuthResult.java

```diff
@@ -0,0 +1,112 @@
+public class MSALRuntimeAuthResult extends Structure implements MSALRuntimeLibrary.AuthResultCallbackData {
+    public LongByReference authParamsHandle;
+    public LongByReference accountHandle;
+    public LongByReference authResultHandle;
+    public MSALRuntimeUtils.StringByReference idToken = new MSALRuntimeUtils.StringByReference(new WString(""));
+    public MSALRuntimeUtils.StringByReference accessToken = new MSALRuntimeUtils.StringByReference(new WString(""));
+    public LongByReference accessTokenExpirationTime;
+    public MSALRuntimeUtils.StringByReference accountId = new MSALRuntimeUtils.StringByReference(new WString(""));//TODO: split AuthenticationResult into SignInResult (id token/account) and TokenResult (access tokens)? Or keep it simple all-in-one like MSAL Java's AuthenticationResult?
+    public MSALRuntimeUtils.StringByReference accountClientInfo = new MSALRuntimeUtils.StringByReference(new WString(""));
+    private MSALRuntimeLibrary MSALRuntime_INSTANCE;
+    private MSALRuntimeError MSALRuntimeErrorHandler;
+
+    public void setMSALRuntimeLibraryInstance (MSALRuntimeLibrary MSALRuntime_INSTANCE) {
+        this.MSALRuntime_INSTANCE = MSALRuntime_INSTANCE;
+    }
+
+    public void setMSALRuntimeErrorHandler (MSALRuntimeLibrary MSALRuntime_INSTANCE) {
+        this.MSALRuntimeErrorHandler = new MSALRuntimeError();
+        this.MSALRuntimeErrorHandler.MSALRuntime_INSTANCE = MSALRuntime_INSTANCE;
+    }
+
+    public void initializeAuthParameters(String clientId, String authority, String scopes, String redirectUri, String claims) {
+
+        System.out.println("===Creating auth parameters");
+        MSALRuntimeErrorHandler.printAndReleaseError(MSALRuntime_INSTANCE.MSALRUNTIME_CreateAuthParameters(new WString(clientId),
+                new WString(authority),
+                authParamsHandle));
+        MSALRuntimeUtils.incrementRefsCounter(MSALRuntime_INSTANCE, this.MSALRuntimeErrorHandler);//reference to an MSALRUNTIME_AUTH_PARAMETERS_HANDLE
+
+        System.out.println("===Setting parameters");
+        MSALRuntimeErrorHandler.printAndReleaseError(MSALRuntime_INSTANCE.MSALRUNTIME_SetRedirectUri(authParamsHandle.getValue(), new WString("https://login.microsoftonline.com/common/oauth2/nativeclient")));
+        MSALRuntimeErrorHandler.printAndReleaseError(MSALRuntime_INSTANCE.MSALRUNTIME_SetRequestedScopes(authParamsHandle.getValue(), new WString(scopes)));
+        MSALRuntimeErrorHandler.printAndReleaseError(MSALRuntime_INSTANCE.MSALRUNTIME_SetRedirectUri(authParamsHandle.getValue(), new WString(redirectUri)));
+        if (claims != null) {
+            MSALRuntimeErrorHandler.printAndReleaseError(MSALRuntime_INSTANCE.MSALRUNTIME_SetDecodedClaims(authParamsHandle.getValue(), new WString(claims)));
+        }
+    }
+
+    public void parseAndReleaseAuthResult_SignInSilently() {
```

Comment:
I'm confused why a class named *Result has API to parseAndReleaseAuthResult_SignInSilently as public API? Same goes for a lot of the other API in this class.

Reply:
As I mentioned in another comment, because of how JNA handles conversion between Java and C/C++, any of the fields in this class that will be referenced in a native context need to be public. To be safe, I just made all of the methods public as well to avoid exceptions during development, but will revisit the scoping once we get some more thorough tests done.